staging/src/k8s.io/apiserver/pkg/storage/cacher/delegator.go

```diff
@@ -35,16 +42,38 @@ import (
 	"k8s.io/klog/v2"
 )
 
+var (
+	// ConsistencyCheckPeriod is the period of checking consistency between etcd and cache.
+	// 5 minutes were proposed to match the default compaction period. It's magnitute higher than
+	// List latency SLO (30 seconds) and timeout (1 minute).
+	ConsistencyCheckPeriod = 5 * time.Minute
+)
+
 func NewCacheDelegator(cacher *Cacher, storage storage.Interface) *CacheDelegator {
-	return &CacheDelegator{
+	d := &CacheDelegator{
 		cacher:  cacher,
 		storage: storage,
+		stopCh:  make(chan struct{}),
+	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.WatchCacheConsistenyChecking) {
```

Comment:
Sorry, maybe I wasn't clear.

I don't think we should use feature-gate for it, because I can't imagine ever graduating that for production clusters (or at least for all clusters).
We should probably use env-var for consistency with other similar mechanisms, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/mutation_detector.go#L35-L37
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/util/consistencydetector/list_data_consistency_detector.go#L31-L33
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/util/consistencydetector/list_data_consistency_detector.go#L31-L33

Reply:
Done